### PR TITLE
Add rustbuild command `bench`

### LINKED ISSUE #37897: Allow building benchmarks with rustbuild
Rustbuild should allow building / testing / running benchmarks

I imagine rustbuild's approach will be a lot better for benchmark development and running compared to the old makefiles.

Optional parts:

- Allow testing different optimization / compiler flags

### PR DESCRIPTION
**Pull request by bluss**
Add command bench to rustbuild, so that `./x.py bench <path>` can compile and run benchmarks.

`./x.py bench --stage 1 src/libcollections` and `./x.py bench --stage 1 src/libstd` should both compile well. Just `./x.py bench` runs all benchmarks for the libstd crates.

Fixes #37897

### CODE REVIEW

File: src/bootstrap/lib.rs [hunk 1]
```diff
@@ -457,7 +458,8 @@ impl Build {
         if self.config.verbose || self.flags.verbose {
             cargo.arg("-v");
         }
-        if self.config.rust_optimize {
+        // FIXME: cargo bench does not accept `--release`
+        if self.config.rust_optimize && cmd != "bench" {
```
Comment (bluss) [inline src/bootstrap/lib.rs]: This was slightly concerning

**Comment by bluss** [comment]
Not sure if I've added bench = false to enough crates to have ./x.py bench complete yet.

**Comment by alexcrichton** [comment]
@bors: r+

Looks great to me, thanks!

**Pull request merged**